### PR TITLE
Feature/jc19 375 search

### PR DESCRIPTION
**Pull request by andrey-gazzaev**
#JC19-375

### CODE REVIEW

File: apps/vanilla/src/pages/index.html [hunk 1]
```diff
@@ -16,6 +16,9 @@
       <option value="status">Status</option>
       <option value="aired">Aired start</option>
     </select>
+
+    <input type="search" class="input-search" placeholder="Find anime by name">
```
Comment (panov-aleksey-saritasa) [inline apps/vanilla/src/pages/index.html:20]: Consider adding `label` tag to improve accessibility

**Review by panov-aleksey-saritasa — APPROVED** [review]

File: apps/vanilla/src/core/utils/api.ts [hunk 2]
```diff
@@ -29,12 +29,16 @@ export interface PaginationConfig {
 
   /** Limit page. */
   readonly limit?: number;
+
+  /** Search line. */
```
Comment (artem-sotnichenko-saritasa) [inline apps/vanilla/src/core/utils/api.ts:33]: Search query or text describes the field better.

File: apps/vanilla/src/core/utils/api.ts [hunk 3]
```diff
@@ -29,12 +29,16 @@ export interface PaginationConfig {
...
+  readonly search?: string;
 }
 
 const configDefault = {
   currentPage: FIRST_PAGE,
   ordering: Ordering.None,
   limit: PAGE_SIZE_DEFAULT,
+  search: '',
```
Comment (artem-sotnichenko-saritasa) [inline apps/vanilla/src/core/utils/api.ts:41]: I suggest create constant for default search. I also think that ordering should have a dedicated constant.

File: apps/vanilla/src/pages/index.html [hunk 4]
```diff
@@ -16,6 +16,12 @@
       <option value="status">Status</option>
       <option value="aired">Aired start</option>
     </select>
+
+    <label class="form-label">
+      <span class="form-label_text">Search</span>
```
Comment (artem-sotnichenko-saritasa) [inline apps/vanilla/src/pages/index.html:21]: It is incorrect BEM. I think you meant `text` to be element of block `form-label`, so it should be `form-label__text`.

**Review by artem-sotnichenko-saritasa — APPROVED** [review]

File: apps/vanilla/src/pages/index.css [hunk 5]
```diff
@@ -51,3 +51,21 @@ th {
...
+.form-label {
+  margin: 0 50px;
+  margin-bottom: 20px;
+  display: flex;
+  flex-direction: column;
+}
+
+.form-label__text {
+  display: block;
+  font-size: var(--font-size-400);
```
Comment (timofei-maximov-saritasa) [inline apps/vanilla/src/pages/index.css:64]: Seems like a very big font size 😆. Let's improve css variable naming

**Review by timofei-maximov-saritasa — APPROVED** [review]

**Review by kristinaSvobodinaSaritasa — APPROVED** [review]

File: apps/vanilla/src/pages/table.ts [hunk 6]
```diff
@@ -114,8 +126,37 @@ export class Table {
...
+      updateAnimeList(this.currentPage, this.currentSorting, this.search);
+    });
+  }
+
+  /**
+   * Changes search string.
+   * @param event Some event.
+   */
+  private handleChangeSearch(event: Event): void {
+    event.preventDefault();
```
Comment (chernodub) [inline apps/vanilla/src/pages/table.ts:139]: Is there a point preventing the default event? As far as I understand, nothing's going to happen anyway

**Review by chernodub — APPROVED** [review]

**Pull request closed without merging**